casper/src/test/scala/coop/rchain/casper/helper/BlockGenerator.scala

```diff
@@ -66,6 +65,25 @@ object BlockGenerator {
     BlockStore[F].put(b.blockHash, updatedBlock) *>
       IndexedBlockDagStorage[F].inject(id, updatedBlock, genesis, false)
   }
+
+  private def computeBlockCheckpointFromDeploys[F[_]: Sync: BlockStore](
+      b: BlockMessage,
+      genesis: BlockMessage,
+      dag: BlockDagRepresentation[F],
+      runtimeManager: RuntimeManager[F]
+  ): F[Either[Throwable, (StateHash, StateHash, Seq[InternalProcessedDeploy])]] =
+    for {
+      parents <- ProtoUtil.unsafeGetParents[F](b)
+
+      deploys = ProtoUtil.deploys(b).flatMap(_.deploy)
+
+      _ = assert(
+        parents.nonEmpty || (parents.isEmpty && b == genesis),
+        "Received a different genesis block."
+      )
+
+      result <- computeDeploysCheckpoint[F](parents, deploys, dag, runtimeManager, System.currentTimeMillis())
```

Comment:
I unserstand that this is inside tests but should not you use `Time[F]` here among other places?

Reply:
Good point, didn't spot the convention in here :+1: